File: frontend_v2/lib/models/guide_model.dart
import 'package:flutter/material.dart';

class GuideModel extends ChangeNotifier {
  String fullName = "";
  String email = "";
  String address = "";
  String nic = "";
  String mobileNumber = "";
  String password = "";
  int age = 0;
  String gender = "Male";

  // Horse details
  String horseName = "";
  String horseBreed = "";
  int horseAge = 0;
  String horseColor = "";
  String horseSpecialNotes = "";

  // Guide profile details
  String profileImagePath = "";
  String bio = "";
  String experience = "";
  List<String> languages = [];

  void updateFullName(String newName) {
    fullName = newName;
    notifyListeners();
  }

  void updateEmail(String newEmail) {
    email = newEmail;
    notifyListeners();
  }

  void updateAddress(String newAddress) {
    address = newAddress;
    notifyListeners();
  }

  void updateNic(String newNic) {
    nic = newNic;
    notifyListeners();
  }

  void updateMobileNumber(String newMobileNumber) {
    mobileNumber = newMobileNumber;
    notifyListeners();
  }

  void updatePassword(String newUpdatePassword) {
    password = newUpdatePassword;
    notifyListeners();
  }

  void updateAge(int newAge) {
    age = newAge;
    notifyListeners();
  }

  void updateGender(String newGender) {
    gender = newGender;
    notifyListeners();
  }

  void updateHorseName(String newName) {
    horseName = newName;
    notifyListeners();
  }

  void updateHorseBreed(String newBreed) {
    horseBreed = newBreed;
    notifyListeners();
  }

  void updateHorseAge(int newAge) {
    horseAge = newAge;
    notifyListeners();
  }

  void updateHorseColor(String newColor) {
    horseColor = newColor;
    notifyListeners();
  }

  void updateHorseSpecialNotes(String newNotes) {
    horseSpecialNotes = newNotes;
    notifyListeners();
  }

  // Add new update methods for profile details
  void updateProfileImage(String path) {
    profileImagePath = path;
    notifyListeners();
  }

  void updateBio(String newBio) {
    bio = newBio;
    notifyListeners();
  }

  void updateExperience(String newExp) {
    experience = newExp;
    notifyListeners();
  }

  void updateLanguages(List<String> newLanguages) {
    languages = newLanguages;
    notifyListeners();
  }
}

File: backend/models/guide.model.js
export default class Guide {
    constructor(data) {
        this.fullName = data.fullName || "";
        this.email = data.email || "";
        this.address = data.address || "";
        this.nic = data.nic || "";
        this.mobileNumber = data.mobileNumber || "";
        this.password = data.password || "";
        this.age = data.age || 0;
        this.gender = data.gender || "Male";

        this.horseName = data.horseName || "";
        this.horseBreed = data.horseBreed || "";
        this.horseAge = data.horseAge || 0;
        this.horseColor = data.horseColor || "";
        this.horseSpecialNotes = data.horseSpecialNotes || "";

        this.profileImagePath = data.profileImagePath || "";
        this.bio = data.bio || "";
        this.experience = data.experience || "";
        this.languages = data.languages || [];
    }
}
